com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.RenderGraph.cs

```diff
@@ -1133,19 +1149,14 @@ class GenerateColorPyramidData
 
         void GenerateColorPyramid(RenderGraph renderGraph, HDCamera hdCamera, TextureHandle inputColor, TextureHandle output, bool isPreRefraction)
         {
-            // Here we cannot rely on automatic pass culling if the result is not read
-            // because the output texture is imported from outside of render graph (as it is persistent)
-            // and in this case the pass is considered as having side effect and cannot be culled.
-            if (isPreRefraction)
+            if (!isPreRefraction)
```

Comment:
This logic here is becoming too complicated IMO.
We have logic outside the function (cf above RenderDistorsion) and also insde the function.
In term of RenderGraph logic, GenerateColorPyramid should do just what it says it does and all logic calling it or not should be outside. I think this would make things much clearer here (both on caller side, and inside the function)

Reply:
fixed